lib/base/networking/apis/campus_backend.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:grpc/grpc.dart';
+
+import '../protocols/grpc/campus.pbgrpc.dart';
+
+
+class CampusService {
+  CampusService._internal();
+  static final CampusService _instance = CampusService._internal();
+
+  factory CampusService() => _instance;
+
+  ///static CampusService instance that we will call when we want to make requests
+  static CampusService get instance => _instance;
```

Comment:
we decided to use get_it for our singleton services, you can find an example that's already in use in the main() function in main.dart